pkg/resource/typed/typed_resource_test.go

```diff
@@ -135,3 +146,40 @@ func TestUnmarshalJSON(t *testing.T) {
 	assert.NotNil(t, res.TypedSpec().Value)
 	assert.Equal(t, "1", res.TypedSpec().Value.Id)
 }
+
+var BenchStore resource.Resource
+
+func TestCopyAllocations(t *testing.T) {
+	res := typed.NewResource[resDefSpec, resDefSpecExtension](
+		resource.NewMetadata("default", "type", "1", resource.VersionUndefined),
+		resDefSpec{
+			Value: &v1alpha1.ResourceDefinitionSpec{
+				ResourceType:     "my_resource_type",
+				DisplayType:      "display_type",
+				DefaultNamespace: "default_namaspace",
+				Aliases:          []string{"alias1", "alias2"},
+				AllAliases:       []string{"alias1", "alias2", "alias3"},
+				PrintColumns: []*v1alpha1.ResourceDefinitionSpec_PrintColumn{
+					{
+						Name:     "name",
+						JsonPath: ".metadata.name",
+					},
+					{
+						Name:     "namespace",
+						JsonPath: ".metadata.namespace",
+					},
+				},
+				Sensitivity: v1alpha1.ResourceDefinitionSpec_NON_SENSITIVE,
+			},
+		},
+	)
+
+	benchRes := testing.Benchmark(func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			BenchStore = res.DeepCopy()
+		}
+	})
+
+	// Report ns, allocs and bytes per op
+	t.Logf("ns/op: %d, allocs/op: %d, bytes/op: %d", benchRes.NsPerOp(), benchRes.AllocsPerOp(), benchRes.AllocedBytesPerOp())
```

Comment:
should we actually assert on allocations here?

Reply:
Done!